engine/src/main/java/org/hibernate/validator/internal/metadata/aggregated/CascadingMetaData.java

```diff
@@ -40,6 +40,16 @@
 	private final TypeVariable<?> typeParameter;
 
 	/**
+	 * The declared container class: it is the one used in the node of the property path.
+	 */
+	private final Class<?> declaredContainerClass;
```

Comment:
Well, I think it's settled considering the last example I gave you? That's what I put in the BVAL issue I created here: https://hibernate.atlassian.net/projects/BVAL/issues/BVAL-631 .

Moreover the last commit fixes the oddity that was bothering me and triggered the question.

(I have the email mostly prepared if needed)